### PR TITLE
checkpoint: remove template checkpoint pod spec

### PR DESCRIPTION
**Pull request by xiang90**
/cc @aaronlevy @derekparker 

This is based on my offline discussion with @aaronlevy. Kubelet will add -xxx as a suffix to the checkpoint pod it starts, so we do not need to have the special name.

### CODE REVIEW

**Comment by xiang90** [comment]
@pbx0 How should I trigger CI to test this?

**Comment by pbx0** [comment]
@xiang90 Its a little awkward, but one way to do it is to first build a checkpoint image from this branch via the internal jenkins job. Then add a commit to this branch that bumps the checkpoint image to use the newly built image in the `pkg/asset/internal/templates.go` file. See https://github.com/kubernetes-incubator/bootkube/pull/201 for an example. 

The test PR will trigger a test that performs a smoke test and another check that does a basic "destruction" test which restarts the master and runs a basic workload.

At somepoint the checkpoint code will be moved to its own repo or we will improve the PR triggers to work around this.

File: cmd/checkpoint/main.go [hunk 1]
```diff
@@ -31,25 +31,15 @@ const (
 	kubeAPIServer = "kube-apiserver"
 )
 
+var podAPIServerMeta = unversioned.TypeMeta{
```
Comment (aaronlevy) [inline cmd/checkpoint/main.go]: s/podAPIServerMeta/podMeta/ -- not specific to apiServer, and we'll need to use it for other checkpointed pods.

**Comment by aaronlevy** [comment]
One minor comment if you don't mind updating.

File: cmd/checkpoint/main.go (hunk 1 above)
Comment (xiang90) [inline cmd/checkpoint/main.go]: I want to make it clear that the pod meta is from API sever (which we simulate here) not kubelet. But if you think this is confusing, I can remove the API server part.

Comment (aaronlevy) [inline cmd/checkpoint/main.go]: sgtm

**Pull request merged**